### PR TITLE
TF-3256 Update index.html cache config

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #3256 

## Reference
- https://docs.flutter.dev/platform-integration/web/initialization

## Resolved
Updating

### CODE REVIEW

**Comment by chibenwa** [comment]
Please further explain the dark magic going on in here.

Cc @guimard

**Comment by tddang-linagora** [comment]
>Please further explain the dark magic going on in here.

4 months ago we updated our workflows to use Flutter 3.22.2. Per https://docs.flutter.dev/platform-integration/web/initialization:
>This page describes APIs that are available in Flutter 3.22 and later.

We updated the Flutter version, but not follow it up with the new cache config that comes with it. This MAY fix it. Need to deploys to dev to test it first.

File: docs/adr/0054-update-flutter-service-worker-initialization.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - Twake Mail web resources is requested everytime the web is loaded, no cache is used <img src="../images/before-flutter-3.22.png">
+
+## Decision
+
+- We update the initialization of Flutter Service Worker according to the new version of Flutter
+- More reference: https://docs.flutter.dev/platform-integration/web/initialization
+
+## Consequences
+
+- Twake Mail web use cache resources to improve performance <img src="../images/after-flutter-3.22.png">
```
Comment (hoangdat) [inline docs/adr/0054-update-flutter-service-worker-initialization.md:22]: IMO, it still not clear about the result of your change with this image. We should mention that: `static files`, `main.dart.js` only re-fetch in hard refresh. Open app, and refresh, not make any request to server side.

**Review by dab246 — APPROVED** [review]
LGTM

**Review by hoangdat — APPROVED** [review]

**Pull request merged**